test/unit/cli-domain-get-mocked-plugins.js

```diff
@@ -2,45 +2,123 @@
+      realpathSync: sinon.stub().returns('/root/'),
+      writeFile: sinon.stub().yields(null, 'ok')
+    }, fs || {});
+
+    var fakePathFunc = function(){
+      return _.toArray(arguments)
+              .map(function(x){
+                return x.replace(/\.\//g, '');
+              })
+              .join('');
+    };
+
+    getMockedPlugins = injectr('../../src/cli/domain/get-mocked-plugins.js', {
+      'fs-extra': fsMock,
+      path: {
+        join: fakePathFunc,
+        resolve: fakePathFunc
+      },
+      '/root/components/dynamic-plugin.js': dynamicPluginModule,
+      '/root/components/not-a-function.js': notAFunctionModule
+    });
+  };
+
   describe('when setting up mocked plugins', function(){
 
+    describe('when oc.json is in both root and component folder', function(){
+
+      var result;
+      var ocJsonComponent = {
+        registries: [],
+        mocks: {
+          plugins: {
+            static: { foo: 1, bar: 2 }
+          }
+        }
+      };
+
+      var readMock = sinon.stub().returns(ocJsonComponent);
+
+      beforeEach(function(){
+        initialise({
+          existsSync: sinon.stub().returns(true),
+          readJsonSync: readMock
+        });
+        result = getMockedPlugins({ log: _.noop }, '/root/components/');
+      });
+
+      it('should use components folder oc.json as default', function(){
+        expect(readMock.calledOnce).to.be.true;
+        expect(readMock.args[0][0]).to.equal('/root/components/oc.json');
+        expect(result.length).to.equal(2);
+      });
+    });
+
+    describe('when oc.json is in root folder', function(){
+
+      var result;
+      var ocJsonComponent = {
+        registries: [],
+        mocks: {
+          plugins: {
+            static: { foo: 1, bar: 2 }
+          }
+        }
+      };
+      var ocJsonRoot = {
+        registries: [],
+        mocks: {
+          plugins: {
+            static: { foo: 1, bar: 2, baz: 3 }
+          }
+        }
+      };
+
+      var readMock = sinon.stub(),
+          existsMock = sinon.stub();
+      
+      readMock.withArgs('/root/components/oc.json').returns(ocJsonComponent);
+      readMock.withArgs('/root/oc.json').returns(ocJsonRoot);
+
+      existsMock.withArgs('/root/components/oc.json').returns(false);
+      existsMock.withArgs('/root/oc.json').returns(true);
+
+      beforeEach(function(){
+        initialise({
+          existsSync: existsMock,
+          readJsonSync: readMock
+        });
+        result = getMockedPlugins({ log: _.noop }, '/root/components/');
+      });
+
+      it('should use root oc.json', function(){
+        expect(result.length).to.equal(3);
+      });
+    });
+
     describe('when oc.json is missing', function(){
       var result;
       beforeEach(function(){
         initialise({ existsSync: sinon.stub().returns(false) });
-        result = getMockedPlugins(console);
+        result = getMockedPlugins(console, '/root/components/');
```

Comment:
I see that now all the tests are passing the path parameter to the `getMockedPlugins` method; I think we should have tests w/o the second parameter as well what do you think?